### ISSUE TITLE
problem with :drop / :create / :reset etc using mysql

### ISSUE THREAD

**Issue by anomaly256** [merb/merb_datamapper#1]
```
when 'mysql'
  user, password, database = config[:username], config[:password], config[:database]
  `mysql -u #{user} #{password ? "-p #{password}" : ''} -e "drop database #{database}"`
else
```

MySQL's commandl ine tool doesn't work that way, you can't specify the password following the -p argument.  Further more, it seems the user specified in database.yml isn't ending up in that user var!  When I try to run any of these rake tasks I get this:

ERROR 1045 (28000): Access denied for user '-p'@'localhost' (using password: NO)

The command line is getting well-munched here..

**Comment by anomaly256** [comment]
Actually I just rechecked the man page, and it says if specifying the password on the commandline, there needs to be /no space/ between -p and the password.  Also, the entry in Merb::Orms::DataMapper.config holding the user name is [:user] not [:username] upon inspection

**Comment by ngollan** [comment]
This should be fixed now, also see Lighthouse ticket #1340. Having that functionality at all for anything but sqlite is however questionable in my opinion.

**Comment by anomaly256** [comment]
Perhaps.  Although I've found that sqlite /does not/ handle multiple threads accessing the database at once very well (or at all) and so I often find myself in a position where my development database still needs to sit on mysql, and having this functionality does shave a lot of time there.
